worklist/views.py

```diff
@@ -189,23 +189,27 @@ def search_task(request, username, worklist_created_by, worklist_name):
 
 
 # View to update the table displaying all tasks of a worklist
-def update_task_table(request, worklist_created_by, worklist_name):
+@with_logged_in_user
+def update_task_table(request, username, worklist_created_by, worklist_name):
     search_term = request.GET.get('search_term', '')
     results = search_task_helper(search_term, worklist_name, worklist_created_by)
 
-    return render(request, 'show-task-table.html', {'tasks': results['tasks']})
+    return render(request, 'show-task-table.html', {'tasks': results['tasks'],
+                                                    'logged_in_user': username})
 
 
 # Update task information when user edits it
 @login_required
-def update_task_info(request, worklist_created_by, worklist_name):
+def update_task_info(request):
```

Comment:
It seems here we update the status of the task without checking that the user is allowed to, as we do that check client-side.

It's not really a big deal in this case, but in general it's a bad idea to only check things on the client-side, as that's usually easy to bypass by, e.g., calling the relevant JavaScript function from the developer console.

Can you also check here, using the same function you use when serving `show-task-table.html`, whether the current user is allowed to edit what they're trying to edit?

Reply:
Agreed but it'll become difficult to check it in server side because we've one function that is for updating all three (status, progress and claimed_by field). One option is to put 'can_modify_status check' on the server side as it is the superset check. But it won't be full-proof as one can modify progress even if the task is open.
So should I create different views for updating these fields individually?

Reply:
I've adjusted the logic in one view only.